src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/MethodCallHandler.java

```diff
@@ -57,7 +57,8 @@ protected IndentLevel getIndentImpl() {
             // we should increase indentation only if this is the first
             // chained method call which was moved to the next line
             else {
-                indentLevel = new IndentLevel(container.getIndent(), getBasicOffset());
+                indentLevel = new IndentLevel(container.getIndent(),
+                    getIndentCheck().getLineWrappingIndentation());
```

Comment:
Based on the if statement, this code can be hit when, what I assume is the start and end, are on the same line. Should line wrapping indentation only be used when we know for sure they are not on the same line and use basic offset if they aren't?

Reply:
You ask me or @romani ?

Reply:
Any that can answer.

Reply:
>Based on the if statement, this code can be hit when, what I assume is the start and end, are on the same line

I think this condition in `if` statement prevent it `areOnSameLine(container.getMainAst(), getMainAst())`

Reply:
But there is an OR as part of the condition, so the OR is hit when they are not on the same line.

Reply:
`areMethodsChained` - method begins on the same level that parent end -> no need additional offset...

But, I think, you are right about `isChainedMethodCallWrapped`
```java
"str".concat("1").concat("2")  // first line
.concat("3")  // <- use ident from first line
.concat("4")  // <- use ident from first line
.concat("5"); // <- use ident from first line
```
And, I think, we need to add in method `isChainedMethodCallWrapped` check for line wrapping. We don't use parent ident if method call without line wrapping
But this situation was before and, I think, this change should be in another PR
What do you think about it?

Reply:
Can we reproduce potential problem in this PR ? To let us judge on possibly to postpone

Reply:
@romani I add two examples. These examples also reproduce in my IDE

Reply:
@alinkov , it is better to communicate in this issue tracker by output of CLI execution, to be absolutely clear on what you mean and how it (will) behave for users.

Reply:
if I interpret your cases properly ...
Looks like we have issue before your fix:
```
$ cat config.xml 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
                        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
        <module name="Indentation">
            <property name="basicOffset" value="2"/>
            <property name="braceAdjustment" value="0"/>
            <property name="caseIndent" value="2"/>
            <property name="throwsIndent" value="4"/>
            <property name="lineWrappingIndentation" value="4"/>
            <property name="arrayInitIndent" value="2"/>
        </module>
  </module>
</module> 

$ cat Test.java 
class Test {
  public void test() {
    //This is identation without warinngs
    "str".concat("1").concat("2")  // first line
    .concat("3")  // <- use ident from first line
    .concat("4")  // <- use ident from first line
    .concat("5"); // <- use ident from first line

    //This is identation without warinngs
    "str"
.concat("1")
.concat("2")
        .concat("3");
  }
}

$ java -jar /var/tmp/checkstyle-8.28-all.jar -c config.xml Test.java 
Starting audit...
Audit done.
```

if I am right, and you do not want to address it your PR, it is ok, please create new issue on this, and please remove your second commit  to not let somebody be confused why such indentation is ok.

Reply:
@romani rebased

Reply:
@rnveach , are you ok to postpone this issue for later (new separate issue) ?

Reply:
Thought that was already decided.

Reply:
Looks like issue is already existing for this - https://github.com/checkstyle/checkstyle/issues/3802

Reply:
I am ok to resolve this conversation.
@rnveach , please review, if you are satisfied, please resolve conversation.

Reply:
> @alinkov , it would be awesome to finish this PR ...
> 
> please answer to
> [#7238 (comment)](https://github.com/checkstyle/checkstyle/pull/7238#discussion_r361879009) if you have problem to explain ... let us know we will help.
> [#7238 (comment)](https://github.com/checkstyle/checkstyle/pull/7238#discussion_r361878486) in addition to rebase I asking you to confirm.

@romani sorry, I was without computer and internet

Reply:
@romani I try to fix #3802 in another PR and add documentation